exist-core/src/test/xquery/xquery3/bang.xql

```diff
@@ -181,3 +181,46 @@ declare
 function bang:element-left-number-attribute-right() {
     <emp id='1'/> ! number(@id)
 };
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-get () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! map:get(., 'a')
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-get () {
+    ([1], [2], [3]) ! array:get(., 1)
+};
+
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-lookup () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! .?a
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-lookup () {
+    ([1], [2], [3]) ! .?1
+};
+
+(:~
+ : parse error needs to be fixed first
+ : https://github.com/eXist-db/exist/issues/1655
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-lookup-implicit-context () {
+    ([1], [2], [3]) ! ?1
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-lookup-implicit-context () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! ?a
```

Comment:
@joewiz Unary lookup (`?a`) should/must work but is technically a separate issue.

Reply:
Looking at my comment the test will be renamed to `bang:map-unary-lookup`